pyam/_aggregate.py

```diff
@@ -181,6 +191,19 @@ def _agg_weight(data, weight, method):
     if not data.droplevel(["variable", "unit"]).index.equals(weight.index):
         raise ValueError("Inconsistent index between variable and weight!")
 
+    if drop_negative is True:
+        if any(i < 0 for i in weight):
+            logger.warning(
+                "Some of the weights are negative. "
+                "All data with negative values will be dropped. "
+                "If you don't want the values to be dropped "
+                "please use the keyword drop_negative=False"
+            )
+            # Drop negative weights
+            for index, value in weight.items():
+                if value < 0:
+                    weight = weight.drop(labels=[index])
```

Comment:
it should be possible to simply do the following
```suggestion
            weight = weight[weight>0]
```